### PR TITLE
test: integration

### PR DESCRIPTION
**Pull request by excaliborr**
closes SPO-46

### CODE REVIEW

File: test/integration/deployment.test.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    assert.isTrue(deploymentAddress !== undefined);
+  });
+
+  it('Should deploy an ownable contract', async function () {
+    const signer = await this.hre.ethers.provider.getSigner();
+
+    const smartAccount = await this.hre.network.provider.request({
+      method: 'sponsored_getSmartAccountAddress',
+      params: [await signer.getAddress()],
+    });
```
Comment (0xAustrian) [inline test/integration/deployment.test.ts:29]: Just as a note: we should add tests for this kind of function. But lets do it in another PR.

File: test/integration/transactions.test.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { assert } from 'chai';
+import { useEnvironment } from '../helpers';
+import { TEST_TOKEN_ABI } from '../test-token-abi';
+import 'dotenv/config';
+
+describe('Integration transactions', function () {
+  useEnvironment('integration');
+
+  it('Should send a transaction', async function () {
+    const tokenAddr = '0x8A59017aF01F5bC6c73CFB6DeD9e162a5fFaA634';
```
Comment (0xAustrian) [inline test/integration/transactions.test.ts:10]: What do you think about adding those address in a constant file? maybe rename `test-token-abi.ts` to `test-constants.ts` or something?

**Review by 0xAustrian — COMMENTED** [review]
🔥

File: package.json [hunk 3]
```diff
@@ -16,7 +16,9 @@
   ],
   "scripts": {
     "format": "yarn run prettier:fix && yarn run lint:fix",
-    "test": "mocha --exit --recursive 'test/**/*.test.ts'",
+    "test": "yarn compile:integration && mocha --exit --recursive 'test/**/*.test.ts'",
```
[the remark this answers is not in the document]
Comment (excaliborr) [reply package.json:19]: Doing it like this makes it faster to run as it doesnt need to rerun everything to execute tests

**Review by 0xAustrian — APPROVED** [review]
